modules/openideal_discussion/openideal_discussion.module

```diff
@@ -0,0 +1,81 @@
+<?php
+
+/**
+ * @file
+ * Contain Openideal discussion hooks.
+ */
+
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Session\AccountInterface;
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ */
+function openideal_discussion_form_node_discussion_form_alter(&$form, FormStateInterface $form_state, $form_id) {
+  // Hide title for the discussion add form.
+  // @todo create the modal for Discussion edit form.
+  if (\Drupal::request()->query->get('_wrapper_format') == 'drupal_bootstrap4_modal') {
+    // @todo make a checks.
+    $title = \Drupal::request()->request->get('dialogOptions')['title'];
+    $form['title']['widget'][0]['value']['#default_value'] = t('In idea: @idea_title', ['@idea_title' => $title]);
+    $form['title']['#attributes']['class'] = 'd-none';
+    $form['body']['widget'][0]['#title_display'] = 'invisible';
+  }
+}
+
+/**
+ * Implements hook_ENTITY_TYPE_create_access().
+ *
+ * Allow to add comments in Idea discussion only for group member with
+ * such roles: Author, Co-author and Expert and the discussion status is open.
+ */
+function openideal_discussion_comment_create_access(AccountInterface $account, array $context, $entity_bundle) {
+  if (($node = \Drupal::routeMatch()->getParameter('node')) && $node->bundle() == 'discussion') {
+    /** @var \Drupal\openideal_idea\OpenidealHelper $helper */
+    $helper = \Drupal::service('openideal_idea.helper');
+    $member = $helper->getGroupMember($account, $node, 'idea-group_node-discussion');
+    $status = (bool) $node->field_status->value;
+    return AccessResult::forbiddenIf(($member && !$member->hasPermission('create group_node:discussion entity') || !$member || !$status));
```

Comment:
There is something wrong with parenthesizes here.
Let's also try to improve readability. Instead of forbidding, can we allow access in certain conditions?
For example:
`AccessResult::allowedIf($status && $member && $member->hasPermission('create group_node:discussion entity'));`

Reply:
Hi, I reworked the logic and now we're using the permission.checker from group modules, please check it out.

Reply:
Looks better. Can we just swap the order then?
!$status || !$member || (...)